File: saturday/week13/zapbu/p2018.py
N = int(input())

l, r = 0, 0
ans, sum = 1, 0

while r < N:
    if sum < N:
        r += 1
        sum += r
    elif sum > N:
        sum -= l
        l += 1
    else:
        ans += 1
        r += 1
        sum += -l + r
        l += 1

print(ans)

File: saturday/week_13/zapbu/p2018.kt
fun main() {
    val N = readln().toInt()

    var l = 0
    var r = 0
    var ans = 1
    var sum = 0

    while (r < N) {

        when {
            sum < N -> {
                r++
                sum += r
            }

            sum > N -> {
                sum -= l
                l++
            }

            else -> {
                ans++

                sum += -l++ + ++r
            }
        }
    }

    println(ans)
}
